samcli/lib/replace_uri/replace_uri.py

```diff
@@ -0,0 +1,94 @@
+from samcli.lib.utils.packagetype import ZIP
+
+
+def _replace_local_codeuri(template):
+    """
+    Replaces the CodeUri in AWS::Serverless::Function and DefinitionUri in AWS::Serverless::Api and
+    AWS::Serverless::HttpApi to a fake S3 Uri. This is to support running the SAM Translator with
+    valid values for these fields. If this in not done, the template is invalid in the eyes of SAM
+    Translator (the translator does not support local paths)
+    """
+
+    all_resources = template.get("Resources", {})
+    global_settings = template.get("Globals", {})
+
+    for resource_type, properties in global_settings.items():
+
+        if resource_type == "Function":
+            if all(
+                [
+                    _properties.get("Properties", {}).get("PackageType", ZIP) == ZIP
+                    for _, _properties in all_resources.items()
+                ]
+                + [_properties.get("PackageType", ZIP) == ZIP for _, _properties in global_settings.items()]
+            ):
+                _update_to_s3_uri("CodeUri", properties)
+
+    for _, resource in all_resources.items():
+
+        resource_type = resource.get("Type")
+        resource_dict = resource.get("Properties", {})
+
+        if resource_type == "AWS::Serverless::Function" and resource_dict.get("PackageType", ZIP) == ZIP:
```

Comment:
Are these (L32 - L50) all the resources that we need to substitute local path with fake s3 uri? If you take a look `samcli/commands/_utils/resources.py`, you will see the list of all resources that can contain local paths. Do they need to be handled as well?

Reply:
I'm not sure. This code was extracted and used from the already existing sam_template_validator. If there are resources that you believe are missing, then they were not implemented before I extracted this code.

Reply:
I believe this comment is a valid comment.
Also I think we need to handle the Image Uri for Lambda functions of PackageType = Image

Reply:
> For some reason I cannot comment on your last comment below: "I believe this comment is a valid comment..."
> Everything in replace_uri.py came from sam_template_validator. It is out of scope for this project for me to implement that feature.

Ok